### PR TITLE
Persona: Deprecate primaryText

### LINKED ISSUE #4772: Standardize on "text", "secondaryText", and "tertiaryText"
A number of components have inconsistent naming conventions for rendering the text.

We have internally discussed and agreed we want to standardize all things:

Proposal:

Text should be rendered with the `text` prop.
Secondary text should be rendered with the `secondaryText` prop.

Work:

Persona: currently has "primaryText". This should be deprecated in favor of "text".
ContextualMenuItem: currently uses "name", deprecate in favor of "text"
Button: currently uses "description", deprecate in favor of "secondaryText"
ComboBox items: currently uses "value", deprecate in favor of "text"

These should be cleaned up, warnings added, and examples updated.

### PR DESCRIPTION
**Pull request by JasonGore**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #4772
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Deprecate Persona's primaryText prop. Add new 'text' prop. Replace all usages in codebase. Add Deprecation guidelines.

#### Focus areas to test

Persona and PersonaCoin's use of primaryText and text props.

### CODE REVIEW

**Comment by dzearing** [comment]
@mtennoe and @Markionium The PR here is one of a few places where we render "primary text", but the prop isn't called "text". We have a few places where the "text" you provide is called different things, so we want to standardize on the "text" prop. (Obviously, `primaryText` will still work, but we want to clean things up for as much predictability as possible. Let us know if you have any concerns.

**Comment by Markionium** [comment]
Seems ok with me, i don't really see any specific reason why we would have any objections. We'll obviously have to migrate over on our side.

**Comment by JasonGore** [comment]
@Markionium Please let me know if you run into any issues whatsoever with backwards compatibility. I've added as many tests on old behavior as possible to catch any issues, so if I miss anything I want to make sure I modify our process and guidelines appropriately. Thanks 😄

File: packages/office-ui-fabric-react/src/components/Persona/PersonaCoin/PersonaCoin.deprecated.text.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+// Prevent warn deprecations from failing test
+const Utilities = require('@uifabric/utilities/lib/warn');
+
+describe('PersonaCoin', () => {
+  beforeEach(() => {
+    setRTL(false);
+  });
+
+  beforeAll(() => {
+    Utilities.warnDeprecations = jest.fn().mockImplementation(() => { /** no impl **/ });
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Persona/PersonaCoin/PersonaCoin.deprecated.text.tsx]: in my testing, commenting line 17-23 doesn't cause the test to fail..

Comment (cliffkoh) [reply packages/office-ui-fabric-react/src/components/Persona/PersonaCoin/PersonaCoin.deprecated.text.tsx]: So the problem is you called the file `.text.tsx`

File: ghdocs/BestPractices/Deprecation.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+      // 'primaryText' is deprecated and 'text' should be used instead
+      this._warnDeprecations({ 'primaryText': 'text' });
+    }
+    ```
+1. warnDeprecations will most likely cause deprecated tests to fail, requiring mocking of warnDeprecations. Please make note to clear mock at end of tests as shown below.
+
+    ```tsx
+    // Prevent warn deprecations from failing test
+    const Utilities = require('@uifabric/utilities/lib/warn');
```
[the remark this answers is not in the document]
Comment (JasonGore) [reply ghdocs/BestPractices/Deprecation.md]: Yeah I think this should also be mockReset and not mockClear

**Comment by mtennoe** [comment]
No objections from my side about doing this. Aligning on consistent coding names and patterns is important, and it only needs a minor change when consuming the updated version

File: packages/office-ui-fabric-react/src/components/Persona/PersonaCoin/PersonaCoin.deprecated.text.tsx (hunk 1 above)
Comment (JasonGore) [reply packages/office-ui-fabric-react/src/components/Persona/PersonaCoin/PersonaCoin.deprecated.text.tsx]: Nice catch, that caught a backwards compatibility bug too. Shows the tests are working as intended 😄

File: ghdocs/BestPractices/Deprecation.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (JasonGore) [reply ghdocs/BestPractices/Deprecation.md]: Does anybody know of a way to spyOn or mock exported modules from index modules like Utilities.ts? I don't like using the full path show above as most test imports are using `import { setRTL } from '../../Utilities';` but I haven't found a workaround yet.

**Pull request merged**